tests/integration/devfile/cmd_devfile_create_test.go

```diff
@@ -87,398 +81,274 @@ var _ = Describe("odo devfile create command tests", func() {
-	// 	It("should only extract the specified path in the sparseContextDir field", func() {
-	// 		contextDevfile := helper.CreateNewContext()
-	// 		helper.Chdir(contextDevfile)
-	// 		devfile := "devfile.yaml"
-	// 		helper.CopyExampleDevFile(filepath.Join("source", "devfiles", "nodejs", "devfile-with-sparseCheckoutDir"), filepath.Join(contextDevfile, devfile))
-	// 		componentNamespace := helper.RandString(6)
-	// 		helper.CmdShouldPass("odo", "create", "--downloadSource", "--project", componentNamespace)
-	// 		expectedFiles := []string{"app.js", devfile}
-	// 		Expect(helper.VerifyFilesExist(contextDevfile, expectedFiles)).To(Equal(true))
-	// 		helper.DeleteDir(contextDevfile)
-	// 	})
-	// })
-
-	Context("When executing odo create with devfile component, --starter flag and subDir has a valid value", func() {
-		It("should only extract the specified path in the subDir field", func() {
-			originalDir := commonVar.Context
-			defer helper.Chdir(originalDir)
-			contextDevfile := helper.CreateNewContext()
-
-			helper.CopyExampleDevFile(filepath.Join("source", "devfiles", "springboot", "devfile-with-subDir.yaml"), filepath.Join(contextDevfile, "devfile.yaml"))
-			helper.Chdir(contextDevfile)
-			helper.Cmd("odo", "create", cmpName, "--project", commonVar.Project, "--starter").ShouldPass()
-
-			pathsToValidate := map[string]bool{
-				filepath.Join(contextDevfile, "java", "com"):                                            true,
-				filepath.Join(contextDevfile, "java", "com", "example"):                                 true,
-				filepath.Join(contextDevfile, "java", "com", "example", "demo"):                         true,
-				filepath.Join(contextDevfile, "java", "com", "example", "demo", "DemoApplication.java"): true,
-				filepath.Join(contextDevfile, "resources", "application.properties"):                    true,
-			}
-
-			pathsNotToBePresent := map[string]bool{
-				filepath.Join(contextDevfile, "src"):  true,
-				filepath.Join(contextDevfile, "main"): true,
-			}
-
-			found := 0
-			notToBeFound := 0
-			err := filepath.Walk(contextDevfile, func(path string, info os.FileInfo, err error) error {
-				if err != nil {
-					return err
-				}
+		It("should successfully create the devfile component with valid file system path", func() {
+			helper.Cmd("odo", "create", "nodejs", "--devfile", devfilePath).ShouldPass()
+		})
 
-				if ok := pathsToValidate[path]; ok {
-					found++
-				}
+		It("should successfully create the devfile component with valid specifies URL path", func() {
+			helper.Cmd("odo", "create", "nodejs", "--devfile", "https://raw.githubusercontent.com/odo-devfiles/registry/master/devfiles/nodejs/devfile.yaml").ShouldPass()
+		})
 
-				if ok := pathsNotToBePresent[path]; ok {
-					notToBeFound++
-				}
-				return nil
+		It("should fail to create the devfile component", func() {
+			By("using an invalid file system path", func() {
+				helper.Cmd("odo", "create", "nodejs", "--devfile", "#@!").ShouldFail()
```

Comment:
I'm pretty sure the `#@!` will be interpreted as a comment and you will get an error `flag needs an argument: --devfile`, but that is not what we want to use. It would be best to use another string for invalid path